ggplot/scales/scale_color.py

```diff
@@ -0,0 +1,278 @@
+Returning the palette you asked for with that many colors
+"""
+
+_MSG_CONTINUOUS_DISTILLER = """\
+Using a discrete colour palette in a continuous scale.
+Consider using type = "seq" or type = "div" instead"
+"""
+# Palette making utilities #
+
+
+def hue_pal(h=(0 + 15, 360 + 15), c=100, l=65, h_start=0, direction=1):
+    """
+    Utility for making hue palettes for color schemes.
+    """
+    c /= 100.
+    l /= 100.
+    hcl = ColorHCL()
+
+    def func(n):
+        y = deepcopy(h)
+        if (y[1] - y[0]) % 360 < 1:
+            y = (y[0], y[1] - 360. / n)
+        hues = []
+        for x in np.linspace(y[0], y[1], n):
+            hue = ((x + h_start) % 360) * direction
+            hues.append(rgb2hex(hcl(hue, c, l)))
+        return hues
+    return func
+
+
+def grey_pal(start=0.2, end=0.8):
+    """
+    Utility for creating discrete grey scale palette
+    """
+    gamma = 2.2
+    ends = ((0.0, start, start), (1.0, end, end))
+    cdict = {'red': ends, 'green': ends, 'blue': ends}
+    grey_cmap = LinearSegmentedColormap('grey', cdict)
+
+    def func(n):
+        colors = []
+        # The grey scale points are linearly separated in
+        # gamma encoded space
+        for x in np.linspace(start**gamma, end**gamma, n):
+            # Map points onto the [0, 1] palette domain
+            x = (x ** (1./gamma) - start) / (end - start)
+            colors.append(rgb2hex(grey_cmap(x)))
+        return colors
+    return func
+
+
+def brewer_pal(type='seq', palette=1):
+    """
+    Utility for making a brewer palette
+    """
+    def _handle_shorthand(text):
+        abbrevs = {
+            "seq": "Sequential",
+            "qual": "Qualitative",
+            "div": "Diverging"
+        }
+        text = abbrevs.get(text, text)
+        return text.title()
+
+    def _number_to_palette(ctype, n):
+        n -= 1
+        palettes = sorted(brewer2mpl.COLOR_MAPS[ctype].keys())
+        if n < len(palettes):
+            return palettes[n]
+
+    def _max_palette_colors(type, palette_name):
+        """
+        Return the number of colors in the brewer palette
+        """
+        if type == 'Sequential':
+            return 9
+        elif type == 'Diverging':
+            return 11
+        else:
+            # Qualitative palettes have different limits
+            qlimit = {"Accent": 8, "Dark": 8, "Paired": 12,
+                      "Pastel1": 9, "Pastel2": 8, "Set1": 9,
+                      "Set2": 8, "Set3": 12}
+            return qlimit[palette_name]
+
+    type = _handle_shorthand(type)
+    if isinstance(palette, int):
+        palette_name = _number_to_palette(type, palette)
+    else:
+        palette_name = palette
+    nmax = _max_palette_colors(type, palette_name)
+
+    def func(n):
+        # Only draw the maximum allowable colors from the palette
+        # and fill any remaining spots with None
+        _n = n if n <= nmax else nmax
+        bmap = brewer2mpl.get_map(palette_name, type, _n)
```

Comment:
brewer2mpl has a namechange, so should get a conditional import :-(

https://github.com/jiffyclub/palettable/wiki